sklearn/pipeline.py

```diff
@@ -37,6 +39,33 @@
 __all__ = ["Pipeline", "FeatureUnion", "make_pipeline", "make_union"]
 
 
+@contextmanager
+def _handle_warnings(estimator):
```

Comment:
I was confused by this name before reading its docstring. Would something like `_raise_or_warn_if_not_fitted`, or `_ensure_fitted_or_warn`, or `_handle_fit_status` be better?